src/MagentoHackathon/Composer/Magento/Deploystrategy/Copy.php

```diff
@@ -30,7 +30,10 @@ public function createDelegate($source, $dest)
 
 
         // Create all directories up to one below the target if they don't exist
-        $destDir = dirname($destPath);
+        $destDir = realpath(dirname($destPath));
```

Comment:
The previous implementation creates `$destDir` if it not exists. The proposed solution will lead to a situation when 
```php
if (!file_exists($destDir)) {
    mkdir($destDir, 0777, true);
}
```
will never be executed (directory does not exist so `realpath` will return `false`). Such change  of behavior will be backward incompatible and lead to dead code.

To solve the described issue with broken symlinks please use `is_link` function to not affect other cases.